### DISCUSSION TITLE
[Question] How to Directly Import URDF and Mesh Files into IsaacLab Without Converting to USD Format?

### DISCUSSION THREAD

**Discussion by yilin404** [isaac-sim/IsaacLab#1585] [Q&A]
Hello everyone,

I currently have a large number of URDF and Mesh files (e.g., .obj, .glb formats) and I would like to import them into IsaacLab for simulation. Since the number of files is quite large, I want to avoid spending excessive time converting them into USD format.

Is there a way to directly import URDF and Mesh files into IsaacLab without the need for conversion to USD? If anyone has insights or suggestions on how to do this, I would greatly appreciate it!

**Comment by sosno** [comment]
Isaac Sim is built using USD.   It's fundamental to how the Omniverse platform is built.  I believe you must use USD.  So unfortunately you will need to convert them.  

I don't use URDF because I use closed loop kinematic robots -- so for that reason I couldn't answer questions related to that.  

My workflow is to save out STEP files from OnShape assemblies.  Then import into isaac sim which automatically converts them to USD.  I then open the USD file and re-structure / re-organize it since the import process tends to create quite a few unnecessary prims in the hierarchy.  I do this manually today but have considered writing a script to go through and restructure it automatically.  

I know I'm not directly answering your question but hopefully this context is helpful anyway.

**Reply by yilin404** [reply]
Thank you for your reply! I really appreciate your help.
